src/main/common/maths.c

```diff
@@ -360,3 +360,21 @@ int16_t qMultiply(fix12_t q, int16_t input) {
 fix12_t  qConstruct(int16_t num, int16_t den) {
     return (num << 12) / den;
 }
+
+// see https://en.wikipedia.org/wiki/Fast_inverse_square_root
+float fast_rsqrt(float number)
+{
+	long i;
+	float x2, y;
+	const float threehalfs = 1.5F;
+
+	x2 = number * 0.5F;
+	y  = number;
+	i  = * ( long * ) &y;
```

Comment:
In order to get this to compile without a warning you'll have to use a union here.

Reply:
done